blog/_src/posts/2017-05-19-scribble-html.md

```diff
@@ -0,0 +1,269 @@
+
+```
+#lang scribble/text
+@(require "templates.rkt")
+
+<!DOCTYPE html>
+<html lang="en">
+  @(header "Home")
+  <body id="pn-top">
+    @(navbar "Home")
+    <div class="jumbotron">
+```
+
+And here is the start of the `scribble/html`'d homepage:
+
+```
+#lang scribble/html
+@require["templates.rkt"]
+
+@doctype{html}
+@html[lang: "en"]{
+  @header{Home}
+    @body[id: "pn-top"]{
+      @navbar{Home}
+      @div[class: "jumbotron"]{
+```
+
+The pages look similar.
+The new one has more @-signs and parentheses, the old one has more `<`-signs
+ and quotes.
+If you were able to edit the old page, you should be able to edit the new page.
+
+The **key improvement** in the new page is that **common mistakes are now
+ compile-time errors**.
+
+- Before, a typo like `<hmtl>` would generate an ugly webpage.
+  After, a typo like `@hmtl` is a syntax error.
+
+- Before, a typo like `<b>....` with no closing tag would generate an ugly webpage.
+  After, a typo like `@b{....` is a syntax error.
+
+Both flavors of error message come with source-code line numbers.
+This is very very helpful.
+
+
+### Small Improvements
+
+#### 1. More Functions
+
+Before, the [Teaching page](http://prl.ccs.neu.edu/teaching.html) contained
+ some interesting HTML for rendering vertical text (look for the word "Semantics" to see how this was used):
+
+```
+<span class="how-to-design-programs">S<br />e<br />m<br />a<br />n<br />t<br />i<br />c<br />s<br /><br /></span>
+```
+
+After, the same text is generated from a function call:
+
+```
+@span[class: "how-to-design-programs"]{@vertical-text{Semantics}}
+```
+
+The `vertical-text` function is simple:
+
+```
+@require[(only-in racket/list add-between)]
+
+@(define (vertical-text . str*)
+   (add-between (string->list (append* str*)) (br)))
+```
+
+
+#### 2. More Structure, Less Boilerplate
+
+Here's part of the old definition of "Ben Greenman" on the [People page](file:///Users/ben/code/racket/website/people.html):
+
+```
+<div class="row pn-person">
+  <div class="col-md-12 pn-row-eq-height">
+    <div class="col-md-3 pn-photo">
+      <div class="img-wrapper">
+        <img src="img/ben_greenman.jpg" title="Ben Greenman" alt="Ben Greenman" />
+      </div>
+    </div>
+    <div class="col-md-9">
+      <div class="col-md-4 pn-contact">
+        <span class="pn-name">Ben Greenman</span><br />
+        Advisor: Matthias Felleisen<br />
+        <a href="mailto:types@"@"ccs.neu.edu">types@"@"ccs.neu.edu</a><br />
+        <a href="http://www.ccs.neu.edu/home/types">www.ccs.neu.edu/home/types</a>
+      </div>
+      .... 9 lines omitted ....
+```
+
+The new definition uses a helper function with keyword arguments for each
+ "field" of the person:
+
+```
+@person[#:name "Ben Greenman"
+        #:title "Advisor: Matthias Felleisen"
+        #:e-mail "types@ccs.neu.edu"
+        #:website "http://ccs.neu.edu/home/types"
+        #:history @list["Joined 2014"]
+        #:img "ben_greenman.jpg"]{
+  I like constructions ...
```

Comment:
Pure nitpick, but it's a bit surprising that you show content here that is not in the version above. Maybe include "I like constructions..." on both sides?